src/components/user/review-point-detail/modal.tsx

```diff
@@ -0,0 +1,148 @@
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import styled from 'styled-components';
+import { Input, Modal } from 'antd';
+import TextArea from 'antd/lib/input/TextArea';
+
+import { Button, P } from '@src/component/atoms';
+import { INDIGO_BLUE, FIRE_RED } from '@src/component/atoms/colors';
+
+import { SignType } from '@src/types';
+import ReviewPointService from '@src/services/ReviewPoint';
+
+function ReviewPointDetailModal({ userId, isModalVisible, isSetModalVisible }) {
+  const router = useRouter();
+
+  const [point, setPoint] = useState(0);
+  const [sign, setSign] = useState<SignType>(null);
+  const [reason, setReason] = useState('');
+  const [detailReason, setDetailReason] = useState('');
+
+  const handleChange = (e) => {
+    const {
+      target: { value, name },
+    } = e;
+
+    switch (name) {
+      case 'point':
+        setPoint(value);
+        break;
+      case 'reason':
+        setReason(value);
+        break;
+      case 'detailReason':
+        setDetailReason(value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSignClick = (value) => {
+    setSign(value);
+  };
+
+  const handleOk = async () => {
+    if (!sign) {
+      alert('적립 또는 차감을 선택해주세요');
+    } else {
+      await ReviewPointService.create({
+        userId,
+        sign,
+        amount: point,
+        ...(reason.length && { reason }),
+        ...(detailReason.length && { detailReason }),
+      });
+      router.reload();
+    }
+
+    isSetModalVisible(false);
+    initField();
+  };
+
+  const handleCancel = () => {
+    isSetModalVisible(false);
+    initField();
+  };
+
+  const initField = () => {
+    setPoint(0);
+    setSign(null);
+    setReason('');
+    setDetailReason('');
+  };
+
+  return (
+    <Modal
+      title="포인트 적립/차감하기"
+      visible={isModalVisible}
+      onOk={handleOk}
+      onCancel={handleCancel}
+      cancelText="취소"
+      okText="저장"
+    >
+      <Row>
+        <Label>수정할 포인트</Label>
+        <Input
+          value={point}
+          onChange={handleChange}
+          name="point"
+          style={{ width: '10rem' }}
+        />
+        <StyledButton
+          title="적립"
+          sign={SignType.Plus}
+          selected={sign === SignType.Plus}
+          onClick={() => handleSignClick(SignType.Plus)}
+        />
+        <StyledButton
+          title="차감"
+          sign={SignType.Minus}
+          selected={sign === SignType.Minus}
+          onClick={() => handleSignClick(SignType.Minus)}
+        />
```

Comment:
```suggestion
        <StyledButton
          title="적립"
          sign={SignType.Plus}
          selected={sign === SignType.Plus}
          onClick={toggleSign(SignType.Plus)}
        />
        <StyledButton
          title="차감"
          sign={SignType.Minus}
          selected={sign === SignType.Minus}
          onClick={toggleSign(SignType.Minus)}
        />
```

1. 이중 화살표 함수를 사용해서 리팩토링할 수 있습니다.
2. map 사용을 시도해보세요~